File: backend/services/identity-service/src/main/java/com/lmspilot/identity/config/DevelopmentSeed.java
package com.lmspilot.identity.config;

import com.lmspilot.identity.domain.*;
import java.time.Instant;
import java.util.*;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.*;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
public class DevelopmentSeed implements ApplicationRunner {
    private final RoleRepository roles;
    private final UserAccountRepository users;
    private final PasswordEncoder encoder;
    private final boolean enabled;

    public DevelopmentSeed(RoleRepository r, UserAccountRepository u, PasswordEncoder e,
                           @Value("${lmspilot.seed-demo:true}") boolean enabled) {
        roles = r;
        users = u;
        encoder = e;
        this.enabled = enabled;
    }

    @Override
    @Transactional
    public void run(ApplicationArguments args) {
        if (!enabled) return;
        ensure("00000000-0000-0000-0000-000000000001", "ADM001", "admin", "Quản trị hệ thống", "ADMIN", true, "admin123");
        ensure("00000000-0000-0000-0000-000000000002", "INS001", "instructor", "Giảng viên mẫu", "INSTRUCTOR", false, "instructor123");
        ensure("00000000-0000-0000-0000-000000000003", "STU001", "student", "Học viên mẫu", "STUDENT", false, "student123");
    }

    private void ensure(String id, String code, String username, String name, String role, boolean protect, String pass) {
        UserAccountEntity u = users.findByUsernameIgnoreCase(username).orElseGet(UserAccountEntity::new);
        RoleEntity r = roles.findByCodeIgnoreCase(role).orElseThrow();
        u.id = UUID.fromString(id);
        u.code = code;
        u.username = username;
        u.fullName = name;
        u.passwordHash = encoder.encode(pass);
        u.roles = new LinkedHashSet<>(Set.of(r));
        u.accountType = protect ? AccountType.SYSTEM_ADMIN : AccountType.USER;
        u.protectedAccount = protect;
        u.mustChangePassword = false;
        u.passwordChangedAt = Instant.now();
        u.status = AccountStatus.ACTIVE;
        u.failedLoginCount = 0;
        u.lockedUntil = null;
        users.save(u);
    }
}

File: backend/services/identity-service/src/main/kotlin/com/lmspilot/identity/config/DevelopmentSeed.kt
package com.lmspilot.identity.config

import com.lmspilot.contracts.DefaultRolePermissions
import com.lmspilot.identity.domain.*
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional
import java.util.UUID

@Component
class DevelopmentSeed(
    private val roles: RoleRepository,
    private val users: UserAccountRepository,
    private val encoder: PasswordEncoder,
    @Value("\${lmspilot.seed-demo:false}") private val enabled: Boolean,
    @Value("\${lmspilot.default-admin-password}") private val password: String,
) : ApplicationRunner {
    private val log = LoggerFactory.getLogger(javaClass)

    @Transactional
    override fun run(args: ApplicationArguments) {
        if (!enabled) return
        val admin = ensureRole("ADMIN", "Quản trị viên", DefaultRolePermissions.ADMIN)
        val instructor = ensureRole("INSTRUCTOR", "Giảng viên", DefaultRolePermissions.INSTRUCTOR)
        val learner = ensureRole("LEARNER", "Học viên", DefaultRolePermissions.LEARNER)
        ensureUser(UUID.fromString("00000000-0000-0000-0000-000000000001"), "ADM001", "admin", "Quản trị hệ thống", admin, AccountType.SYSTEM_ADMIN, true)
        ensureUser(UUID.fromString("00000000-0000-0000-0000-000000000002"), "INS001", "instructor", "Giảng viên mẫu", instructor)
        ensureUser(UUID.fromString("00000000-0000-0000-0000-000000000003"), "LEA001", "learner", "Học viên mẫu", learner)
        log.warn("Development seed is enabled. Change temporary passwords immediately outside local development.")
    }

    private fun ensureRole(code: String, name: String, permissions: Set<String>): RoleEntity =
        roles.findByCodeIgnoreCase(code) ?: roles.save(RoleEntity(code = code, name = name, permissions = permissions.toMutableSet(), systemRole = true))

    private fun ensureUser(id: UUID, code: String, username: String, fullName: String, role: RoleEntity, accountType: AccountType = AccountType.USER, protectedAccount: Boolean = false) {
        if (users.findByUsernameIgnoreCase(username) == null) {
            users.save(UserAccountEntity(id = id, code = code, username = username, fullName = fullName, passwordHash = encoder.encode(password), roles = mutableSetOf(role), accountType = accountType, protectedAccount = protectedAccount, mustChangePassword = !protectedAccount, passwordChangedAt = java.time.Instant.now()))
        }
    }
}
